### PR TITLE
Improve concat gather

### PR DESCRIPTION
**Pull request by scxiao**

### CODE REVIEW

File: src/include/migraphx/op/concat.hpp [hunk 1]
```diff
@@ -18,7 +18,7 @@ namespace op {
 
 struct concat
 {
-    std::size_t axis = 0;
+    int axis = 0;
```
Comment (pfultz2) [inline src/include/migraphx/op/concat.hpp]: Do we usually make these attributes `int64_t`?

File: src/tf/tf.cpp [hunk 2]
```diff
@@ -357,7 +357,7 @@ struct tf_parser
     {
         // get index for axis within args
         size_t axis_idx = attributes.at("N").i();
-        size_t axis     = args[axis_idx]->eval().at<int64_t>();
+        int axis        = args[axis_idx]->eval().at<int64_t>();
```
Comment (pfultz2) [inline src/tf/tf.cpp]: Can you write this with `auto`?

File: src/tf/tf.cpp [hunk 3]
```diff
@@ -664,8 +664,7 @@ struct tf_parser
             args.end(),
             std::back_inserter(unsqueezed_args),
             [&](instruction_ref arg) { return prog.add_instruction(op::unsqueeze{{axis}}, arg); });
-        return to_nhwc(
-            prog.add_instruction(op::concat{static_cast<size_t>(axis)}, unsqueezed_args));
+        return to_nhwc(prog.add_instruction(op::concat{static_cast<int>(axis)}, unsqueezed_args));
```
Comment (pfultz2) [inline src/tf/tf.cpp]: Is the `static_cast` necessary?

File: src/include/migraphx/op/concat.hpp (hunk 1 above)
Comment (kahmed10) [reply src/include/migraphx/op/concat.hpp]: yes, normally they are int64 if the axis is negative.

File: src/tf/tf.cpp (hunk 3 above)
Comment (scxiao) [reply src/tf/tf.cpp]: removed

File: src/include/migraphx/op/concat.hpp (hunk 1 above)
Comment (scxiao) [reply src/include/migraphx/op/concat.hpp]: changed

File: src/tf/tf.cpp (hunk 2 above)
Comment (scxiao) [reply src/tf/tf.cpp]: fixed

**Pull request merged**